Documentation/ExtensionArchitecture/Tutorials/Tea/DirectoryStructure.rst

```diff
@@ -0,0 +1,189 @@
+    ├── Tests
+    ├── composer.json
+    ├── ext_emconf.php
+    ├── ...
+    └── README.md
+
+Directory :file:`Classes`
+-------------------------
+
+The folder :file:`Classes` has to contain all PHP classes provided by the
+extension. Otherwise they are not available in the
+:ref:`autoloading <autoload>`.
+
+See also the general chapter on the folder :ref:`extension-classes`.
+
+In the :file:`composer.json` we defined that all PHP classes should
+automatically be loaded from the directory :file:`Classes`:
+
+..  tabs::
+
+    ..  group-tab:: Composer
+
+        ..  code-block:: json
+            :caption: EXT:tea/composer.json, extract
+            :emphasize-lines: 4
+
+            {
+                "autoload": {
+                    "psr-4": {
+                        "TTN\\Tea\\": "Classes/"
+                    }
+                }
+            }
+
+    ..  group-tab:: Legacy
+
+        ..  code-block:: php
+            :caption: EXT:tea/ext_emconf.php, extract
+            :emphasize-lines: 4
+
+            $EM_CONF[$_EXTKEY] = [
+                'autoload' => [
+                    'psr-4' => [
+                        'TTN\\Tea\\' => 'Classes/',
+                    ],
+                ],
+            ];
+
+Further this line defines the namespace all PHP classes in this directory
+have to be situated in to be found by the :ref:`PSR-4 autoloading <autoload>`.
+
+The folder :file:`Classes` contains several subfolders:
+
+..  code-block:: none
+    :caption: Directory structure of EXT:tea
+
+    $ tree typo3conf/ext/tea
+    ├── Classes
+        ├── Controller
+        ├── Domain
+        |   ├── Model
+        |   └── Repository
+        └──  ViewHelpers
+
+Extbase is based on the pattern Model-View-Controller (MVC). And you can already
+find directories for the **model** and the **controller** here.
+
+In most cases the **view** is handled by classes provided by the framework
+and configured via templating. Therefore there is no folder for the view as a
+whole.
+
+Additional logic needed for the view can be provided by **ViewHelpers** and
+should be stored in the according folder.
+
+..  note::
+    ViewHelpers are a feature of the Fluid templating engine. See also the
+    chapter :ref:`fluid-custom-viewhelper`.
+
+Directory :file:`Configuration`
+-------------------------------
+
+See also the general chapter on the folder :ref:`extension-configuration-files`.
+
+The folder :file:`Configuration` contains several subfolders:
+
+..  code-block:: none
+    :caption: Directory structure of EXT:tea
+
+    $ tree typo3conf/ext/tea
+    ├── Configuration
+        ├── FlexForms
+        ├── TCA
+        |   └── Overrides
+        ├── TsConfig
+        |   ├── Page
+        |   └── User
+        ├── TypoScript
+        |   ├── constants.typoscript
+        |   └── setup.typoscript
+        └──  Services.yaml
+
+:file:`Configuration/FlexForms`
+    Contains the configuration of additional input fields to
+    configure the plugin.
```

Comment:
```suggestion
    configure plugins.
```
Without switchableControllerActions you might have several plugins now. Would also like to see a link to FlexForm explanation.